pkg/commands/exec_live_default.go

```diff
@@ -0,0 +1,132 @@
+// +build !windows
+
+package commands
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"sync"
+	"unicode/utf8"
+
+	"github.com/kr/pty"
+)
+
+// RunCommandWithOutputLiveWrapper runs a command and return every word that gets written in stdout
+// Output is a function that executes by every word that gets read by bufio
+// As return of output you need to give a string that will be written to stdin
+// NOTE: If the return data is empty it won't written anything to stdin
+// NOTE: You don't have to include a enter in the return data this function will do that for you
+func RunCommandWithOutputLiveWrapper(c *OSCommand, command string, output func(string) string) (errorMessage string, codeError error) {
+	cmdOutput := []string{}
+
+	splitCmd := ToArgv(command)
+	cmd := exec.Command(splitCmd[0], splitCmd[1:]...)
+
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, "LANG=en_US.utf8", "LC_ALL=en_US.UTF-8")
+
+	tty, err := pty.Start(cmd)
+
+	if err != nil {
+		return errorMessage, err
+	}
+
+	// canAsk makes sure there are no data races in go
```

Comment:
I think it is possible to reduce the lines of code here by using a channel rather than a mutex.

Here's a little program I made to demonstrate how you'd go about achieving this:
```
package main

import (
	"fmt"
	"time"
)

func main() {
	c := make(chan struct{})

	go func() {
		time.Sleep(2 * time.Second)
		c <- struct{}{}
	}()

loop:
	for {
		time.Sleep(250 * time.Millisecond)
		select {
		case <-c:
			fmt.Println("program finished")
			break loop
		default:
			fmt.Println("program still not finished")
		}
	}
}

```

The idea is that you make a channel who type is the empty struct{} because these take up zero space. Then you say 'in two seconds, send an empty struct object to this channel'. Then start a loop that selects on the channel's value to be read, and in the default block you can do whatever you want. Once the other goroutine has sent the empty struct object to the channel, we break out of the loop.

So you could adapt that like so:
```
		loop: for scanner.Scan() {
                         select {
                           case <-c:
                             break loop
                         }
                         default {
				toOutput := re.ReplaceAllString(scanner.Text(), "")
				cmdOutput = append(cmdOutput, toOutput)
				toWrite := output(toOutput)
				if len(toWrite) > 0 {
					_, _ = tty.Write([]byte(toWrite + "\n"))
				}
                         }
		}
```

then below, 
```
	if err = cmd.Wait(); err != nil {
		c <- struct{}{}
```

let me know your thoughts

Reply:
Oke to me..  
if it has less code and does the same thing as a mutex why not.  
As i'm reading into channel it doesn't looks like mutexes are made for these kinds of things they are made to handle more data than only 1 boolean